x/bsonx/bsoncore/iterator.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsoncore
+
+import (
+	"errors"
+	"io"
+
+	"go.mongodb.org/mongo-driver/bson/bsontype"
+)
+
+// ErrCorruptedDocument is returned when a full document couldn't be read from
+// the sequence.
+var ErrCorruptedDocument = errors.New("invalid DocumentSequence: corrupted document")
```

Comment:
This error variable isn't used outside of this package and doesn't need to be exported.